File: common/src/main/java/juuxel/adorn/item/group/ItemGroupingOption.java
package juuxel.adorn.item.group;

import juuxel.adorn.util.Displayable;
import net.minecraft.network.chat.Component;

public enum ItemGroupingOption implements Displayable {
    BY_MATERIAL("by_material"),
    BY_SHAPE("by_shape");

    private final Component displayName;

    ItemGroupingOption(String id) {
        displayName = Component.translatable("gui.adorn.item_grouping_option." + id);
    }

    @Override
    public Component getDisplayName() {
        return displayName;
    }
}

File: common/src/main/kotlin/juuxel/adorn/item/group/ItemGroupingOption.kt
package juuxel.adorn.item.group

import juuxel.adorn.util.Displayable
import net.minecraft.text.Text

enum class ItemGroupingOption(val id: String) : Displayable {
    BY_MATERIAL("by_material"),
    BY_SHAPE("by_shape");

    override val displayName: Text = Text.translatable("gui.adorn.item_grouping_option.$id")
}
